src/lib_json/json_reader.cpp

```diff
@@ -18,6 +18,8 @@
 #include <memory>
 #include <set>
 #include <limits>
+#include <stdio.h>
```

Comment:
Minimally, wouldn't these have to be `cstring` and `cstdio` for strict compatibility with modern compilers?

Adding headers is fine. But could you tell us where this was failing to compile?